### ISSUE TITLE
Get-IntuneWin32App cannot find some apps when tenant has > 1000 apps

### ISSUE THREAD

**Issue by averhaegen** [MSEndpointMgr/IntuneWin32App#78]
Hi,

I experienced an issue using the **Get-IntuneWin32App -Displayname 'xyz'**, could not find some of my apps in the tenant.
I did some troubleshooting and I noticed it is because the command will only load the first 1000 apps that are in my tenant and then compare if the display name is found.

I tested this by using the Get-IntuneWin32App and outputting all the apps it had found. Only the first 1000 apps in my tenant were returned. (Apps were alphabetically sorted, and in my case, it returned only apps with starting letter 'A' to 'S', while the app I was looking for started with a 'Z'..😃 ) .

As a possible solution, I think you can maybe rewrite the function and use an odata filter when executing the Graph request. (see: https://learn.microsoft.com/en-us/graph/filter-query-parameter?tabs=http)
Or another method is to maybe check if there are more than 1000 apps and execute a second graph call to retrieve the next 1000 apps and so on.. ? 

Here is a direct link to the command in your code:
https://github.com/MSEndpointMgr/IntuneWin32App/blob/84b3d83a7e8454c3f27909017a67cf917e556747/Public/Get-IntuneWin32App.ps1#L60

Thanks and kind regards,

Alexander

**Comment by NickolajA** [comment]
You're absolutely right, the code doesn't currently handle pagination, at all really. I'll make sure this is added in 1.4.1.

**Comment by ashley-almeida** [comment]
Hi,

This is still open. Can you fix it in the next version?
